### ISSUE TITLE
CI slaves are missing Android build-tools

### ISSUE THREAD

**Issue by philwo** [bazelbuild/continuous-integration#21]
The CI machines are missing build-tools in their Android SDK in /home/ci/android:

```
philwo@ubuntu-14-04-slave-2:~$ ls -l /home/ci/android/android-sdk-linux/
total 24
drwxrwxr-x  2 ci 5000 4096 Aug 14  2015 add-ons
drwxrwxr-x 20 ci 5000 4096 Aug 18 16:02 platforms
drwxr-xr-x  5 ci root 4096 Aug 18 16:00 platform-tools
-rw-rw-r--  1 ci 5000 1158 Aug 14  2015 SDK Readme.txt
drwxr-xr-x  2 ci root 4096 Aug 18 16:02 temp
drwxr-xr-x 11 ci root 4096 Aug 18 16:02 tools
```

This causes our Android integration tests to fail with this error:

```
ERROR: /home/ci/.cache/bazel/_bazel_ci/a761298a0949227106f79c62c3bebb6e/bazel-sandbox/e8411b06-bba6-4da4-8436-bf269a6c38b3-118/_tmp/android_integration_test_1/root/413f4715efc88c54ccd6e46318cdc885/external/androidsdk/BUILD:5:1: @androidsdk//:sdk: invalid label 'build-tools//mainDexClasses.rules' in attribute 'main_dex_classes' in 'android_sdk' rule: invalid target name 'build-tools//mainDexClasses.rules': target names may not contain '//' path separators.
ERROR: /home/ci/.cache/bazel/_bazel_ci/a761298a0949227106f79c62c3bebb6e/bazel-sandbox/e8411b06-bba6-4da4-8436-bf269a6c38b3-118/_tmp/android_integration_test_1/root/413f4715efc88c54ccd6e46318cdc885/external/androidsdk/BUILD:5:1: @androidsdk//:sdk: missing value for mandatory attribute 'main_dex_classes' in 'android_sdk' rule.
ERROR: /home/ci/.cache/bazel/_bazel_ci/a761298a0949227106f79c62c3bebb6e/bazel-sandbox/e8411b06-bba6-4da4-8436-bf269a6c38b3-118/_tmp/android_integration_test_1/root/413f4715efc88c54ccd6e46318cdc885/external/androidsdk/BUILD:5:1: Traceback (most recent call last):
	File "/home/ci/.cache/bazel/_bazel_ci/a761298a0949227106f79c62c3bebb6e/bazel-sandbox/e8411b06-bba6-4da4-8436-bf269a6c38b3-118/_tmp/android_integration_test_1/root/413f4715efc88c54ccd6e46318cdc885/external/androidsdk/BUILD", line 5
		create_android_sdk_rules(name = "androidsdk", build_tools_v... "", <2 more arguments>)
	File "/home/ci/.cache/bazel/_bazel_ci/a761298a0949227106f79c62c3bebb6e/bazel-sandbox/e8411b06-bba6-4da4-8436-bf269a6c38b3-118/_tmp/android_integration_test_1/root/413f4715efc88c54ccd6e46318cdc885/external/bazel_tools/tools/android/android_sdk_repository_template.bzl", line 180, in create_android_sdk_rules
		native.filegroup(name = "build_tools_libs", srcs = ...]))
	File "/home/ci/.cache/bazel/_bazel_ci/a761298a0949227106f79c62c3bebb6e/bazel-sandbox/e8411b06-bba6-4da4-8436-bf269a6c38b3-118/_tmp/android_integration_test_1/root/413f4715efc88c54ccd6e46318cdc885/external/bazel_tools/tools/android/android_sdk_repository_template.bzl", line 182, in native.filegroup
		native.glob(["build-tools/%s/lib/**" % build...])
empty segment not permitted.
```

I think our gce/linux-android.sh script is currently not working correctly, because I see this error when I run its "android sdk update" command manually:

```
ci@ubuntu-14-04-slave:~/android/android-sdk-linux$ tools/android update sdk --no-ui -t tools,platform,platform-tool
Refresh Sources:
  Fetching https://dl.google.com/android/repository/addons_list-2.xml
  Validate XML
  Parse XML
  Fetched Add-ons List successfully
  Refresh Sources
  Fetching URL: https://dl.google.com/android/repository/repository-11.xml
  Validate XML: https://dl.google.com/android/repository/repository-11.xml
  Parse XML:    https://dl.google.com/android/repository/repository-11.xml
  Fetching URL: https://dl.google.com/android/repository/addon.xml
  Validate XML: https://dl.google.com/android/repository/addon.xml
  Parse XML:    https://dl.google.com/android/repository/addon.xml
  Fetching URL: https://dl.google.com/android/repository/glass/addon.xml
  Validate XML: https://dl.google.com/android/repository/glass/addon.xml
  Parse XML:    https://dl.google.com/android/repository/glass/addon.xml
  Fetching URL: https://dl.google.com/android/repository/extras/intel/addon.xml
  Validate XML: https://dl.google.com/android/repository/extras/intel/addon.xml
  Parse XML:    https://dl.google.com/android/repository/extras/intel/addon.xml
  Fetching URL: https://dl.google.com/android/repository/sys-img/android/sys-img.xml
  Validate XML: https://dl.google.com/android/repository/sys-img/android/sys-img.xml
  Parse XML:    https://dl.google.com/android/repository/sys-img/android/sys-img.xml
  Fetching URL: https://dl.google.com/android/repository/sys-img/android-wear/sys-img.xml
  Validate XML: https://dl.google.com/android/repository/sys-img/android-wear/sys-img.xml
  Parse XML:    https://dl.google.com/android/repository/sys-img/android-wear/sys-img.xml
  Fetching URL: https://dl.google.com/android/repository/sys-img/android-tv/sys-img.xml
  Validate XML: https://dl.google.com/android/repository/sys-img/android-tv/sys-img.xml
  Parse XML:    https://dl.google.com/android/repository/sys-img/android-tv/sys-img.xml
  Fetching URL: https://dl.google.com/android/repository/sys-img/google_apis/sys-img.xml
  Validate XML: https://dl.google.com/android/repository/sys-img/google_apis/sys-img.xml
  Parse XML:    https://dl.google.com/android/repository/sys-img/google_apis/sys-img.xml
Error: Ignoring unknown package filter 'tools'
Warning: The package filter removed all packages. There is nothing to install.
         Please consider trying to update again without a package filter.
```

Investigating.

**Comment by philwo** [comment]
@ahumesky FYI

**Comment by damienmg** [comment]
That means they changed the filter for tools, let's look at `android list sdk --all --extended`

**Comment by philwo** [comment]
It looks like we also have to pass "--all" to the update command:

ci@ubuntu-14-04-slave:~/android/android-sdk-linux$ ./tools/android update sdk --no-ui --all --filter tools

Installing Archives:
  Preparing to install archives
  Downloading Android SDK Tools, revision 25.1.7
  Installing Android SDK Tools, revision 25.1.7
    Installed Android SDK Tools, revision 25.1.799%)
  Done. 1 package installed.

**Comment by philwo** [comment]
However, we will need to explicitly install build-tools-22.0.1, because that's what we use for CI (and I recently noticed that Bazel doesn't work with the newest versions anyway).

**Comment by philwo** [comment]
I'm sending a fix.

**Comment by damienmg** [comment]
I am pretty sure it worked on 24.0.0 last time I tried.

**Comment by damienmg** [comment]
22.0.1 is the default version, it does work with 24.0.0 and detect it.

**Comment by philwo** [comment]
Oh, I didn't know we aim to be compatible with always the latest version. In that case of course we should not hardcode the version, I agree :) Modified and submitted the fix.

**Comment by philwo** [comment]
OK, the fix for this is a bit more complicated.

Here's what we do at the moment:
echo y | (cd android-sdk-linux && tools/android update sdk --no-ui --all -t tools,platform,platform-tool)

**Comment by philwo** [comment]
OK this didn't help and the real fix for this is a bit more complicated.

Here's what we do at the moment:

```
echo y | (cd android-sdk-linux && tools/android update sdk --no-ui --all -t tools,platform,platform-tool)
```

This has the following issues:
- Will not work if there is more than one license prompt.
- Does not install build-tools (because we explicitly install only SDK tools, the platforms and the platform-tools).

Why can't we just add "build-tools" to the filter or something similar? Because build-tools has no "latest version" label - they're all explicitly versioned:

```
ci@ubuntu-15-10-slave:~/android/android-sdk-linux$ tools/android list sdk --all --extended | fgrep -i build-tools
id: 4 or "build-tools-24.0.1"
id: 5 or "build-tools-24.0.0"
id: 6 or "build-tools-23.0.3"
id: 7 or "build-tools-23.0.2"
id: 8 or "build-tools-23.0.1"
id: 9 or "build-tools-23.0.0"
id: 10 or "build-tools-22.0.1"
id: 11 or "build-tools-22.0.0"
[...]
```

So, we could just grep the ID of the latest version out of that list and add it to the filter list, but there seems to be an easier fix: Just drop all of the filters. This installs build-tools, platforms, platform-tools, SDK tools and a bunch of other things.

I'll send a new fix that also uses "expect" to correctly "yes" all the license prompts.

**Comment by damienmg** [comment]
Maybe just fix to version 24.0.0 then?
